sdk/maintenance/Azure.ResourceManager.Maintenance/src/Custom/RestOperations/MaintenanceConfigurationsRestOperations.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager.Maintenance.Models;
+
+namespace Azure.ResourceManager.Maintenance
+{
+    internal partial class MaintenanceConfigurationsRestOperations
+    {
+        internal HttpMessage CreateDeleteRequest(string subscriptionId, string resourceGroupName, string resourceName)
+        {
+            var message = _pipeline.CreateMessage();
+            var request = message.Request;
+            request.Method = RequestMethod.Delete;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/subscriptions/", false);
+            uri.AppendPath(subscriptionId, true);
+            uri.AppendPath("/resourcegroups/", false);
+            uri.AppendPath(resourceGroupName, true);
+            uri.AppendPath("/providers/Microsoft.Maintenance/maintenanceConfigurations/", false);
+            uri.AppendPath(resourceName, true);
+            uri.AppendQuery("api-version", _apiVersion, true);
+            request.Uri = uri;
+            request.Headers.Add("Accept", "application/json");
+            _userAgent.Apply(message);
+            return message;
+        }
+
+        /// <summary> Delete Configuration record. </summary>
+        /// <param name="subscriptionId"> Subscription credentials that uniquely identify a Microsoft Azure subscription. The subscription ID forms part of the URI for every service call. </param>
+        /// <param name="resourceGroupName"> Resource Group Name. </param>
+        /// <param name="resourceName"> Maintenance Configuration Name. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="subscriptionId"/>, <paramref name="resourceGroupName"/> or <paramref name="resourceName"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="subscriptionId"/>, <paramref name="resourceGroupName"/> or <paramref name="resourceName"/> is an empty string, and was expected to be non-empty. </exception>
+        public async Task<Response<MaintenanceConfigurationData>> DeleteAsync(string subscriptionId, string resourceGroupName, string resourceName, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(subscriptionId, nameof(subscriptionId));
+            Argument.AssertNotNullOrEmpty(resourceGroupName, nameof(resourceGroupName));
+            Argument.AssertNotNullOrEmpty(resourceName, nameof(resourceName));
+
+            MaintenanceConfigurationData data = Get(subscriptionId, resourceGroupName, resourceName, cancellationToken).Value;
```

Comment:
I think we shouldn't add an extra API call here, can we directly return null instead of the data? Returning null is also consistent with the service actual behavior.

Reply:
I added comments for why customize this operation:

// In the swagger this delete operation is defined to returen a MaintenanceConfigurationData model, but actually the service doesn't return, that cause exception when try to descerialize the response payload .
// As this lib has already GAed, so here using custom code to retrieve the data before running the delete operation that can keep backward compatibility.

Reply:
This is making an extra REST API call?
We should not do this.
How about we just return `null` for the data, or construct one using our internal constructor by setting minimum properties of it so that we do not have to call the GET API

Reply:
I saw an extra REST API call in our recording for the delete. We should avoid it.

Reply:
I still think we should avoid the extra call. Think about this scenario, what if the customer call Delete to delete a non-existed resource. We will get the exception first from the Get method, this might confuse the customer.

Reply:
Fixed, use directive to fix the swagger spec and customize the `Delete` operations in the `MaintenanceConfigurationResource`.